gnovm/stdlibs/testing/testing.go

```diff
@@ -2,13 +2,100 @@ package testing
 
 import (
 	"errors"
+
+	gno "github.com/gnolang/gno/gnovm/pkg/gnolang"
+	"github.com/gnolang/gno/gnovm/stdlibs/std"
+	teststd "github.com/gnolang/gno/gnovm/tests/stdlibs/std"
+	"github.com/gnolang/gno/tm2/pkg/crypto"
 )
 
 func X_unixNano() int64 {
 	// only implemented in testing stdlibs
 	return 0
 }
 
+func X_testSetContext(
+	m *gno.Machine,
+	isOrigin bool,
+	originCaller string,
+	originPkgAddress string,
+	currRealmAddr string, currRealmPkgPath string,
+	origSendDenoms []string, origSendAmounts []int64,
+	origSpendDenoms []string, origSpendAmounts []int64,
+	chainID string,
+	height int64,
+	timeUnix int64, timeNano int64,
+) {
+	ctx := m.Context.(*teststd.TestExecContext)
+
+	if chainID != "" {
+		ctx.ChainID = chainID
+	}
```

Comment:
I think instead of checking whether the values are the zero value, we should always set these values with what is passed in from the VM